includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -2119,6 +2119,11 @@ public function send_request( Request $request ) {
 	 * @throws API_Exception - If the account ID hasn't been set.
 	 */
 	protected function request( $params, $api, $method, $is_site_specific = true, $use_user_token = false, bool $raw_response = false ) {
+		if ( ! $this->is_api_request_allowed( $api ) ) {
+			Logger::info( "Platform API request '$api' wasn't executed. API requests are only performed if the account is connected." );
+			return []; // this will result as 200 OK with empty array response.
```

Comment:
I am not too happy or confident about returning 200 OK. It doesn't seem like the right thing to do.

To my mind, the caller needs to be informed that the request wasn't allowed rather than tricking it that it went through and there were no results.

I vote for throwing an `API_Exception` with a 403 HTTP response code because the request shouldn't have happened in the first place (someone didn't check the account state before calling).

I see no problem in throwing an exception and responding with a 403. The caller should be able to handle it - all public wrapper methods (e.g. `WC_Payments_API_Client::get_payment_method`) declare that they throw `API_Exception`.

WDYT @marcinbot? Is there something I am missing here?

Reply:
>  the API client shouldn't be responsible for checking the account status (the circular dependency on the account class is an indicator of this). It also takes control away from the callers. The API client should just be responsible for sending out authenticated HTTP requests.

totally agree with you

> would you be ok with closing this PR and https://github.com/Automattic/woocommerce-payments/issues/9883 without shipping it?

in this case yes, we can close this issue 

thanks for the feedback @marcinbot